app/src/test/java/waiter/CommunicatorTest.java

```diff
@@ -0,0 +1,21 @@
+package waiter;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CommunicatorTest {
+
+    @Test
+    void establishesCommunication() throws IOException {
+        String userInput = "Doesn't matter whats here, simulating any user input message";
+        ListenerMock listenerMock = new ListenerMock(userInput);
+        MessengerMock messengerMock = new MessengerMock();
```

Comment:
Suggestion: Moving the mock classes to their corresponding class to so that it's easier to update them later on.

Reply:
Like have the mock classes and the corresponding class in the same file?

Reply:
Not the tests. I meant moving the mocks to be adjacent to their actual class for example,
```
// inside main/java/waiter/ClientConnection/ directory

ClientConnection.java
mock/

// inside the mock/ directory
ClientConnectionMock.java
```

Reply:
Hmmm that be cool, I guess my only question is why I would include the mocks in the src directory when they only really exist in the test code/directory, I appreciate all the help 😄

Reply:
Good question. I'll use `ClientConnection` and `ClientConnectionMock` as an example but it should apply to all classes and their mocks. This colocating is symbolic in a way that a mock is highly coupled to the actual class. For example if you update `ClientConnection.someMethodCall`, you would also need to update the mock to reflect that change. Similarly if you add a new method in ClientConnection, you'll likely need to update the mock as well. You can create a directory for all the mocks in the test directory as well but I think  colocating the mock and actual class makes updating a little easier.

Reply:
That does indeed make alot of sense, the fact that they are so heavily coupled is a good reason to keep them together, I might make that a separate PR as its more of an organizational thing, but I will definitely update later! If that sounds all good to you 😄 , I can also do it in this PR I just thought it would be better in its own.

Reply:
👍 sounds good either way. Let's make sure we capture this somewhere if you're planning on doing this in a future PR.

Reply:
Would a refactor/future PR ticket in trello be a good idea? I have it in my notes but I figured we'd want it captured somewhere we can all see it, plus I have a couple of future PR's I want to do so I can put those in tickets as well if this sounds good

Reply:
@jonmanahan Thinking about this a bit more – even though this change pertains more to code organization it only really includes files that were introduced in this PR. Making the change now won't make the PR more difficult to comprehend for reviewers. I'm leaning more towards having you do this now.

I've made comments in the past about separating changes like – file naming, location changes, updated directory hierarchies – into a separate PR from feature development, but this feels different to me since the files you are organizing here were introduced as part of this feature.

Reply:
"Making the change now won't make the PR more difficult to comprehend for reviewers." I like this point and it makes alot of sense, I can see how the fact that the files were introduced in this feature makes it make sense to include this change in the current PR, as opposed to separating just because its more organization. I am also leaning towards making the change in this PR

Reply:
Check for resolved in [this commit](https://github.com/jonmanahan/waiter/pull/2/commits/b5c07b226a5e771c655351044d9ef186473cd6eb), if it resolves the issues please hit resolve conversation if possible, thank you all a ton!